dgc/DgCertificate.cs

```diff
@@ -0,0 +1,276 @@
+namespace DGC
+{
+    using System;
+    using System.Collections.Generic;
+
+    using System.Globalization;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+
+    /// <summary>
+    /// EU Digital Green Certificate
+    /// </summary>
+    public partial class DgCertificate
+    {
+        /// <summary>
+        /// Date of Birth of the person addressed in the DGC. ISO 8601 date format restricted to
+        /// range 1900-2099
+        /// </summary>
+        [JsonProperty("dob")]
+        public DateTimeOffset DateOfBirth { get; set; }
+
+        /// <summary>
+        /// Surname(s), given name(s) - in that order
+        /// </summary>
+        [JsonProperty("nam")]
+        public Nam Name { get; set; }
+
+        /// <summary>
+        /// Recovery Group
+        /// </summary>
+        [JsonProperty("r", NullValueHandling = NullValueHandling.Ignore)]
+        public RecoveryElement[] Recovery { get; set; }
+
+        /// <summary>
+        /// Test Group
+        /// </summary>
+        [JsonProperty("t", NullValueHandling = NullValueHandling.Ignore)]
+        public TestEntry[] Test { get; set; }
+
+        /// <summary>
+        /// Vaccination Group
+        /// </summary>
+        [JsonProperty("v", NullValueHandling = NullValueHandling.Ignore)]
+        public VaccinationEntry[] Vaccination { get; set; }
+
+        /// <summary>
+        /// Version of the schema, according to Semantic versioning (ISO, https://semver.org/ version
+        /// 2.0.0 or newer)
+        /// </summary>
+        [JsonProperty("ver")]
+        public string Version { get; set; }
+    }
+
+    /// <summary>
+    /// Surname(s), given name(s) - in that order
+    ///
+    /// Person name: Surname(s), given name(s) - in that order
+    /// </summary>
+    public partial class Nam
+    {
+        /// <summary>
+        /// The family or primary name(s) of the person addressed in the certificate
+        /// </summary>
+        [JsonProperty("fn", NullValueHandling = NullValueHandling.Ignore)]
+        //[JsonConverter(typeof(PurpleMinMaxLengthCheckConverter))]
+        public string FamilyName { get; set; }
+
+        /// <summary>
+        /// The family name(s) of the person transliterated
+        /// </summary>
+        [JsonProperty("fnt")]
```

Comment:
https://github.com/ehn-digital-green-development/ehn-dgc-schema/blob/main/examples/vac.json has it as fnt or am I misunderstanding something

Reply:
> https://github.com/ehn-digital-green-development/ehn-dgc-schema/blob/main/examples/vac.json has it as fnt or am I misunderstanding something

No you're not and you're quite right - they were recently changed to fnt and gnt instead of _fn and _gn. Perfect. Leave as-is.